src/mono/mono/mini/interp/transform.c

```diff
@@ -8159,23 +8221,171 @@ generate_compacted_code (TransformData *td)
 	g_ptr_array_free (td->relocs, TRUE);
 }
 
+static void
+interp_mark_reachable_bblocks (TransformData *td)
+{
+	InterpBasicBlock **queue = mono_mem_manager_alloc0 (td->mem_manager, td->bb_count * sizeof (InterpBasicBlock*));
+	InterpBasicBlock *current;
+	int cur_index = 0;
+	int next_position = 0;
+
+	// FIXME There is no need to force eh bblocks to remain alive
+	current = td->entry_bb;
+	while (current != NULL) {
+		if (current->eh_block || current->patchpoint_data) {
+			queue [next_position++] = current;
+			current->reachable = TRUE;
+		} else {
+			current->reachable = FALSE;
+		}
+		current = current->next_bb;
+	}
+
+	queue [next_position++] = td->entry_bb;
+	td->entry_bb->reachable = TRUE;
+
+	// We have the roots, traverse everything else
+	while (cur_index < next_position) {
+		current = queue [cur_index++];
+		for (int i = 0; i < current->out_count; i++) {
+			InterpBasicBlock *child = current->out_bb [i];
+			if (!child->reachable) {
+				queue [next_position++] = child;
+				child->reachable = TRUE;
+			}
+		}
+	}
+}
+
+static gboolean
+interp_prev_ins_defines_var (InterpInst *ins, int var1, int var2)
+{
+	// Check max of 5 instructions
+	for (int i = 0; i < 5; i++) {
+		ins = interp_prev_ins (ins);
+		if (!ins)
+			return FALSE;
+		if (mono_interp_op_dregs [ins->opcode] && (ins->dreg == var1 || ins->dreg == var2))
+			return TRUE;
+	}
+	return FALSE;
+}
+
+static void
+interp_reorder_bblocks (TransformData *td)
+{
+	InterpBasicBlock *bb;
+
+	for (bb = td->entry_bb; bb != NULL; bb = bb->next_bb) {
+		InterpInst *first = interp_first_ins (bb);
+		if (!first)
+			continue;
+		if (MINT_IS_CONDITIONAL_BRANCH (first->opcode)) {
+			// This means this bblock has a single instruction, the conditional branch
+			int i = 0;
+			int lookup_var2 = (mono_interp_op_dregs [first->opcode] > 1) ? first->sregs [1] : -1;
+			while (i < bb->in_count) {
+				InterpBasicBlock *in_bb = bb->in_bb [i];
+				InterpInst *last_ins = interp_last_ins (in_bb);
+				if (last_ins && last_ins->opcode == MINT_BR && interp_prev_ins_defines_var (last_ins, first->sregs [0], lookup_var2)) {
+					// This bblock is reached unconditionally from one of its parents
```

Comment:
Does it mean that it could save a `mov` operation per BB with unconditional reach?

Reply:
explained here - https://github.com/dotnet/runtime/pull/78015/commits/5379908ff9bcac8cc4b8ae5fb22a9684b81d280c